### DISCUSSION TITLE
Emulation without OS

### DISCUSSION THREAD

**Discussion by medovy** [qilingframework/qiling#1360] [General]
Can Qiling's non-OS emulation approach be an alternative to wine/proton?

**Comment by sskras** [comment]
I am interested in the idea behind your question.

If you want to execute Windows binaries, they will need an implementation of user-mode DLLs, at least main ones:
- ntdll.dll
- kernel32.dll
- gdi32.dll
- user32.dll
- advapi32.dll
- ... and other primary ones.

Windows OS has one implementation of them.  Wine has another implementation.  Where do you plan to take these libraries from?

**Reply by medovy** [reply]
So qiling is unable to run MS Windows binaries?
